### PR TITLE
documentation fix #335 and examples added for the Azure VNET peering and AWS Private Link

### LINKED ISSUE #335: Typo in documentation
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
# Copy-paste your version info here
```

### Terraform Configuration File 

```hcl
# Copy-paste your configuration info here
```

### References

https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/private_endpoint_interface_link

data source and resource names are wrong here.

### PR DESCRIPTION
**Pull request by nikhil-mongo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
documentation fix #335 and examples added for the Azure VNET peering and AWS Private Link

### CODE REVIEW

**Comment by themantissa** [comment]
@nikhil-mongo I'll review this as is but in the future:
1) PRs should be one topic at a time.  Note how the submissions from DigitialOnUs are broken out. This allows one to keep code changes separate.  This should be 3 PRs.  One for each example, one for the doc fix.
2) When a PR is ready request a review from me.  If it's one we need more on I'll add other reviewers.

File: examples/aws-atlas-privatelink/Readme.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Example - AWS and Atlas PrivateLink with Terraform
+
+This project aims to provide a very straight-forward example of setting up PrivateLink connection between AWS and MongoDB Atlas.
+
+
+## Dependencies
+
+* Terraform v0.12 or v0.13
```
Comment (leofigy) [inline examples/aws-atlas-privatelink/Readme.md]: question, the provider works for terraform v0.12, but the example is for v0.13, given the versions.tf, would it not better to say the example was tested with v0.13? like in https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/examples/Atlas_Database_Users ?

Comment (nikhil-mongo) [reply examples/aws-atlas-privatelink/Readme.md]: Updated Readme.md with correct Terraform version matching the versions.tf

**Review by leofigy — APPROVED** [review]
LGTM, please wait for Melissa's feedback

File: examples/aws-atlas-privatelink/atlas-cluster.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+resource "mongodbatlas_cluster" "cluster-atlas" {
+  project_id                   = var.atlasprojectid
+  name                         = "cluster-atlas"
+  num_shards                   = 1
+  replication_factor           = 3
+  provider_backup_enabled      = true
+  auto_scaling_disk_gb_enabled = true
+  mongo_db_major_version       = "4.2"
```
Comment (themantissa) [inline examples/aws-atlas-privatelink/atlas-cluster.tf:8]: Just for next time we may want to start using 4.4 in examples.

File: examples/aws-atlas-privatelink/atlas-cluster.tf [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  provider_encrypt_ebs_volume = true
+  provider_instance_size_name = "M10"
+  provider_region_name        = var.atlas_region
+}
+output "atlasclusterstring" {
+  value = mongodbatlas_cluster.cluster-atlas.connection_strings
+}
+output "plstring" {
+  value = lookup(mongodbatlas_cluster.cluster-atlas.connection_strings[0].aws_private_link_srv, aws_vpc_endpoint.ptfe_service.id)
+}
```
Comment (themantissa) [inline examples/aws-atlas-privatelink/atlas-cluster.tf:24]: Most linters will comment if you don't have a final newline, not blocking but fyi.

File: examples/aws-atlas-privatelink/aws-vpc.tf [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+/*Security-Group
+Ingress - Port 80 -- limited to instance
+          Port 22 -- Open to ssh without limitations
+Egress  - Open to All*/
+
+resource "aws_security_group" "primary_default" {
+  name_prefix = "default-"
+  description = "Default security group for all instances in ${aws_vpc.primary.id}"
+  vpc_id      = aws_vpc.primary.id
+  ingress {
```
Comment (themantissa) [inline examples/aws-atlas-privatelink/aws-vpc.tf:45]: I would lean towards providing more locked down examples or add a comment to be sure to narrow ranges, blocks, etc.  Fine for now but if you'd like to submit a 2nd PR based on this input that would be fine.  No rush.

File: examples/aws-atlas-privatelink/variables.tf [hunk 5]
```diff
@@ -0,0 +1,32 @@
+variable "public_key" {
+  description = "The public API key for MongoDB Atlas"
```
Comment (themantissa) [inline examples/aws-atlas-privatelink/variables.tf:2]: Normally when I include any secrets I remind folks it's better to use a secret manager, etc and to be sure to git ignore any variables files in as many places as possible.  Again not a blocker but a nice-to-do for, especially new, users

File: website/docs/d/private_endpoint_interface_link.html.markdown [hunk 6]
```diff
@@ -8,7 +8,7 @@ description: |-
 
 # mongodbatlas_private_endpoint_link
 
-`mongodbatlas_private_endpoint_link` describe a Private Endpoint Link. This represents a Private Endpoint Link Connection that wants to retrieve details in an Atlas project.
+`mongodbatlas_private_endpoint_interface_link` describe a Private Endpoint Link. This represents a Private Endpoint Link Connection that wants to retrieve details in an Atlas project.
```
Comment (themantissa) [inline website/docs/d/private_endpoint_interface_link.html.markdown:11]: lol I fixed the resource and missed the datasource!  Nice catch, ty!

**Review by themantissa — APPROVED** [review]
@nikhil-mongo left a few non blocking comments for next time you submit a PR.  A great start of examples and appreciated!

**Pull request merged**